src/expr.rs

```diff
@@ -558,41 +553,9 @@ impl PyExpr {
             }
         })
     }
-
-    pub fn column_name(&self, plan: PyLogicalPlan) -> PyResult<String> {
-        self._column_name(&plan.plan()).map_err(py_runtime_err)
-    }
 }
 
 impl PyExpr {
```

Comment:
@jdye64 you may want to review this PR since it removes code that I  believe you originally added

Reply:
@jdye64 - I had removed the method because it relied on `DFField` which was removed in `datafusion`.

The last commit attempts to re-implement the method using arrow's `Field`. 

I'd still much appreciate any feedback / context!

Reply:
also cc @charlesbluca

Reply:
It looks like Dask SQL is using a pinned version of this repo from more than six months ago, so we likely won't get a review from the team right away. The new functionality based on `Field` looks good to me, so I will go ahead and merge this PR.

Reply:
Yeah this is fine. Honestly we need to come up with a better way to get the column name anyway and as you mentioned are using a pinned older version for now anyway.